app/scripts/services/tile-proxy.js

```diff
@@ -117,25 +91,17 @@ const throttleAndDebounce = (func, interval, finalWait) => {
     reset();
   };
 
-  debounced.immediate = () => {
-    // @ts-expect-error - Does not pass in `args` to function. Fine for functions without additional args... but not type safe!
-    func({
-      sessionId,
-      requests: bundledRequest,
-    });
-  };
-
   let wait = false;
   /**
-   * @param {WrappedTilesRequest} request
+   * @param {T} item
    * @param {Args} args
    */
-  const throttled = (request, ...args) => {
-    bundleRequests(request);
+  const throttled = (item, ...args) => {
+    items.push(item);
```

Comment:
Here, we just push a new item to our queue anytime the caller makes a request.

Reply:
These (this one and the ones above) notes are helpful. I would add them as comments in the code so they're persisted.